### PR TITLE
core/state: fix staticcheck warnings

### PR DESCRIPTION
**Pull request by fjl**
Also remove dependency on gopkg.in/check.v1 in tests.
This removes the following staticcheck warnings:

```
core/state/dump.go:58:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/dump.go:62:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/dump.go:66:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/dump.go:82:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/dump.go:88:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/dump.go:137:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/dump.go:146:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/dump.go:156:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/journal.go:131:3: field prev is unused (U1000)
core/state/journal.go:132:3: field prevDirty is unused (U1000)
core/state/statedb.go:127:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/statedb.go:133:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/statedb.go:139:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/statedb.go:158:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/statedb.go:169:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/statedb.go:173:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/statedb.go:182:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/statedb.go:192:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/statedb.go:197:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/statedb.go:204:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/statedb.go:214:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/statedb.go:220:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/statedb.go:226:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/statedb.go:234:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/statedb.go:244:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/statedb.go:249:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/statedb.go:253:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/statedb.go:261:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/statedb.go:276:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/statedb.go:285:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/statedb.go:294:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/statedb.go:301:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/statedb.go:312:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/statedb.go:321:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/statedb.go:327:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/statedb.go:336:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/statedb.go:349:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/statedb.go:357:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/statedb.go:364:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/statedb.go:371:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/statedb.go:378:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/statedb.go:385:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/statedb.go:394:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/statedb.go:406:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/statedb.go:493:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/statedb.go:498:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/statedb.go:508:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/statedb.go:532:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/statedb.go:570:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/statedb.go:630:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/statedb.go:638:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/statedb.go:654:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
core/state/statedb.go:713:7: receiver name should be a reflection of its identity; don't use generic names such as "this" or "self" (ST1006)
```

### CODE REVIEW

File: core/state/dump.go [hunk 1]
```diff
@@ -134,17 +137,17 @@ func (self *StateDB) dump(c collector, excludeCode, excludeStorage, excludeMissi
 }
 
 // RawDump returns the entire state an a single large object
-func (self *StateDB) RawDump(excludeCode, excludeStorage, excludeMissingPreimages bool) Dump {
+func (db *StateDB) RawDump(excludeCode, excludeStorage, excludeMissingPreimages bool) Dump {
```
Comment (karalabe) [inline core/state/dump.go]: In the other file you used `s` for the short name. The checker will complain that there are 2 names.

Comment (fjl) [reply core/state/dump.go]: Fixing

File: core/state/journal.go [hunk 2]
```diff
@@ -127,9 +127,7 @@ type (
 		hash common.Hash
 	}
 	touchChange struct {
-		account   *common.Address
-		prev      bool
```
Comment (holiman) [inline core/state/journal.go]: What? We can remove those?

Comment (holiman) [reply core/state/journal.go]: Oh, right, I changed that

https://github.com/ethereum/go-ethereum/commit/958ed4f3d977b08465915e475e11aaab3d2dc574

**Pull request merged**